src/install-update.ts

```diff
@@ -0,0 +1,167 @@
+  } catch (err) {
+    updateResp.error = { message: err.message };
+  }
+
+  return updateResp;
+}
+
+/**
+ * updateDependencyFile reads the contents of a provided path, calls updateDependencyMap
+ * to swap out the current versions with the latest releases available, and then writes
+ * to the dependency file. Returns a summary of updates made.
+ */
+export async function updateDependencyFile(
+  path: string,
+  releases: Release[],
+): Promise<Update[]> {
+  try {
+    const dependencyMap = await getJSON(path);
+    const { imports, hooks } = dependencyMap;
+
+    // If file doesn't exist or expected dependency key is missing
+    if (!imports && !hooks) return [];
+
+    const dependencyKey = imports ? "imports" : "hooks";
+
+    // Update only the dependency-related key in given file ("imports" or "hooks")
+    const { updatedDependencies, updateSummary } = updateDependencyMap(
+      dependencyMap[dependencyKey],
+      releases,
+    );
+
+    // Replace the dependency-related section with the updated version
+    dependencyMap[dependencyKey] = updatedDependencies;
+    await Deno.writeTextFile(
+      path,
+      JSON.stringify(dependencyMap, null, 2).concat("\n"),
+    );
+
+    return updateSummary;
+  } catch (_) {
+    throw new Error(`An error occurred while reading a dependency file`);
+  }
+}
+
+/**
+ * updateDependencyMap takes in a map of the dependencies' key/value pairs and,
+ * if an update exists for a dependency of the same name in the releases provided,
+ * replaces the existing version with the latest version of the dependency. Returns
+ * an updated map of all dependencies, as well as an update summary of each.
+ */
+export function updateDependencyMap(
+  dependencyMap: { [key: string]: string },
+  releases: Release[],
+) {
+  const updatedDependencies: { [key: string]: string } = { ...dependencyMap };
+  const updateSummary: Update[] = [];
+
+  // Loop over key, val pairs of 'imports' or 'hooks', depending on file provided
+  for (const [key, val] of Object.entries(updatedDependencies)) {
+    for (const { name, current, latest } of releases) {
+      // If the dependency matches an available release,
+      // and an update is available, replace the version
+      if (current && latest && val.includes(name)) {
+        updatedDependencies[key] = updatedDependencies[key].replace(
+          current,
+          latest,
+        );
+        updateSummary.push({
+          name,
+          current,
+          latest,
+        });
+      }
+    }
+  }
+
+  return { updatedDependencies, updateSummary };
+}
+
+/**
+ * runBuildHook runs the `build` hook found in mod.ts. In the context of
+ * `install-update`, this is in order to cache the dependency version
+ * updates that occurred.
+ */
+function runBuildHook(): void {
+  try {
+    const { hooks: { build } } = projectScripts([]);
+    const buildArgs = build.split(" ");
+    Deno.run({ cmd: buildArgs });
+  } catch (_) {
```

Comment:
Same comment here, I think this exception may contain valuable information to the user / future maintainers (us) triaging potential exceptions being thrown in unforeseen circumstances! So my preference would be to extract the error message from the exception here and include it in the new error being re-thrown.

Reply:
Sounds good! Since we surface these errors to the user, the approach I'll take is if no errors have occurred previously (with the actual installations), we attempt to build/cache. If an error occurs, we surface/return that error as the top-level error of the response.